packages/javascript/bh-shared-ui/src/components/CreateUserForm/CreateUserForm.tsx

```diff
@@ -14,426 +14,699 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-import { Button } from '@bloodhoundenterprise/doodleui';
 import {
-    Alert,
+    Button,
+    Card,
     Checkbox,
     DialogActions,
-    DialogContent,
+    DialogClose,
+    DialogTitle,
+    Form,
     FormControl,
-    FormControlLabel,
-    FormHelperText,
-    Grid,
-    InputLabel,
-    MenuItem,
+    FormField,
+    FormItem,
+    FormLabel,
+    FormMessage,
+    Input,
     Select,
-    SelectChangeEvent,
-    TextField,
-} from '@mui/material';
-import { CreateUserRequest, SSOProvider } from 'js-client-library';
-import React, { useEffect } from 'react';
-import { Controller, useForm } from 'react-hook-form';
+    SelectContent,
+    SelectItem,
+    SelectPortal,
+    SelectTrigger,
+    SelectValue,
+    Tooltip,
+} from '@bloodhoundenterprise/doodleui';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Alert } from '@mui/material';
+import { CreateUserRequest, Environment, EnvironmentRequest, Role, SSOProvider } from 'js-client-library';
+import { Minus } from 'lucide-react';
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
 import { useQuery } from 'react-query';
 import { MAX_EMAIL_LENGTH, MAX_NAME_LENGTH, MIN_NAME_LENGTH } from '../../constants';
-import { useListDisplayRoles } from '../../hooks';
+import { useAvailableEnvironments } from '../../hooks/useAvailableEnvironments/useAvailableEnvironments';
 import { apiClient } from '../../utils';
 
-export type CreateUserRequestForm = Omit<CreateUserRequest, 'SSOProviderId'> & { SSOProviderId: string | undefined };
+export type CreateUserRequestForm = Omit<CreateUserRequest, 'sso_provider_id'> & {
+    sso_provider_id: string | undefined;
+};
 
 const CreateUserForm: React.FC<{
-    onCancel: () => void;
-    onSubmit: (user: CreateUserRequestForm) => void;
-    isLoading: boolean;
     error: any;
-}> = ({ onCancel, onSubmit, isLoading, error }) => {
-    const {
-        control,
-        handleSubmit,
-        setValue,
-        formState: { errors },
-        setError,
-    } = useForm<CreateUserRequestForm>({
-        defaultValues: {
-            emailAddress: '',
-            principal: '',
-            firstName: '',
-            lastName: '',
-            password: '',
-            needsPasswordReset: false,
-            roles: [1],
-            SSOProviderId: '',
+    isLoading: boolean;
+    onSubmit: (user: CreateUserRequestForm) => void;
+    showEnvironmentAccessControls?: boolean;
+}> = ({ error, isLoading, onSubmit, showEnvironmentAccessControls }) => {
+    const defaultValues = {
+        email_address: '',
+        principal: '',
+        first_name: '',
+        last_name: '',
+        password: '',
+        needs_password_reset: false,
+        roles: [3],
```

Comment:
This seems like another use case for a hook/util/function that would map UserRoles to ids so we could be certain that we are initiating the array with the Read-Only user role. 

Unfortunately that would mean we cant completely define the `defaultValues` until the `useListDisplayRoles` request completed. I think in that case we would have to initiate roles as an empty array (looks like the form field is in a loading state until the `getRolesQuery` completes anyway, yay!), and pass an `onSuccess` handler to the `useQuery` options that would do something like `form.setValue('',[userRoleMap.ReadOnly?.id])`